### PR TITLE
cli, examples, tests: Remove global JavaScript dependencies in favour of local

### LINKED ISSUE #382: Stop using global JS dependencies
It's a bit hard to run JS tests b/c we need to install dependencies globally. Can we pin JS dependencies with a lock file and use `package.json`?

### PR DESCRIPTION
**Pull request by tomlinton**
The global JS dependencies seemed to be causing a host of problems and there are a lot of open issues so this removes them. This shouldn't break any existing setups because npx will use the global if local isn't available.

- Give each test/example its own package.json with explicit dependencies
- Use npx to run mocha/ts-mocha
- Updated the cli to use npx
- Update the docs
- The escrow test reference the ts package by path. I changed it to the published package as it's inconsistent with the rest and had mixed dependencies in its package.json. 

Closes https://github.com/project-serum/anchor/issues/382, https://github.com/project-serum/anchor/issues/982, https://github.com/project-serum/anchor/issues/957, https://github.com/project-serum/anchor/issues/717, https://github.com/project-serum/anchor/issues/638, https://github.com/project-serum/anchor/issues/437

### CODE REVIEW

File: docs/src/getting-started/installation.md [hunk 1]
```diff
@@ -53,12 +45,6 @@ On Linux systems you may need to install additional dependencies if `cargo insta
...
 ```
 
-To install the JavaScript package.
-
-```bash
-npm install -g @project-serum/anchor
-```
-
 Make sure your `NODE_PATH` is set properly so that globally installed modules
 can be resolved.
```
Comment (fanatid) [inline docs/src/getting-started/installation.md:49]: Since we will not have globally installed packages, can we remove this line?

**Comment by fanatid** [comment]
So many `version` fields now... I think we need a script that updates them all, otherwise, we always will miss something.

Did you tried workspace? https://github.com/project-serum/anchor/issues/948 (this should help to remove a lot of lock files)

File: docs/src/getting-started/installation.md (hunk 1 above)
Comment (tomlinton) [reply docs/src/getting-started/installation.md]: Missed that one, thanks!

**Comment by tomlinton** [comment]
Agreed, there is an active issue for a version bump script https://github.com/project-serum/anchor/issues/153. Bumping the JavaScript dependencies will also be a pain.

Workspaces will work, although the flow becomes a little weird, e.g.

```
cd examples/tutorial
npm install
cd basic-0
anchor test
```

instead of:

```
cd examples/tutorial/basic-0
npm install
anchor test
```

(although the second option will still work with workspaces, it'll generate a new lockfile).

It isn't much work to rejig this to use workspaces though.

**Comment by tomlinton** [comment]
Put workspaces in and updated CI to use them. I think it is nice since it's completely optional and won't break the standard `npm install` in each individual example if people want to use that.

**Comment by fanatid** [comment]
@tomlinton thanks a lot that investigated the workspace proposal! Did not know about #153, nice that an issue already exists.

**Comment by fanatid** [comment]
@tomlinton why did you change everything from `npm` to `yarn`?

**Comment by tomlinton** [comment]
@armaniferrante suggested it in Discord and I prefer it to npm. Open for discussion though!

**Comment by fanatid** [comment]
:thinking: how `yarn` is better than `npm`? (I used `yarn` a few years ago when lock files did not exist but now `npm` also use them)

**Comment by tomlinton** [comment]
It used to be faster, have a better lockfile structure and have more useful commands like `yarn why` and `yarn upgrade-interactive`. I'm not sure if npm has now caught up, I never had a reason to move back to npm and yarn just became muscle memory.

**Comment by tomlinton** [comment]
[Here is a branch](https://github.com/tomlinton/anchor/tree/tomlinton/js-dependency-update-npm) that is the same with this but uses npm and npx. I'm not married to either setup.

**Pull request merged**